x-pack/plugins/rule_registry/server/utils/create_get_summarized_alerts_fn.ts

```diff
@@ -0,0 +1,359 @@
+  };
+};
+
+const getQueryByExecutionUuid = (executionUuid: string, ruleId: string, action?: string) => {
+  const filter: QueryDslQueryContainer[] = [
+    {
+      term: {
+        [ALERT_RULE_EXECUTION_UUID]: executionUuid,
+      },
+    },
+    {
+      term: {
+        [ALERT_RULE_UUID]: ruleId,
+      },
+    },
+  ];
+  if (action) {
+    filter.push({
+      term: {
+        [EVENT_ACTION]: action,
+      },
+    });
+  }
+
+  return {
+    body: {
+      size: MAX_ALERT_DOCS_TO_RETURN,
+      track_total_hits: true,
+      query: {
+        bool: {
+          filter,
+        },
+      },
+    },
+  };
+};
+
+interface GetAlertsByTimeRangeOpts {
+  start: Date;
+  end: Date;
+  ruleId: string;
+  ruleDataClientReader: IRuleDataReader;
+  isLifecycleAlert: boolean;
+}
+
+const getAlertsByTimeRange = async ({
+  start,
+  end,
+  ruleId,
+  ruleDataClientReader,
+  isLifecycleAlert,
+}: GetAlertsByTimeRangeOpts) => {
+  if (isLifecycleAlert) {
+    return getLifecycleAlertsByTimeRange({ start, end, ruleId, ruleDataClientReader });
+  }
+
+  return getPersistentAlertsByTimeRange({ start, end, ruleId, ruleDataClientReader });
+};
+
+interface GetAlertsByTimeRangeHelperOpts {
+  start: Date;
+  end: Date;
+  ruleId: string;
+  ruleDataClientReader: IRuleDataReader;
+}
+
+enum AlertTypes {
+  NEW = 0,
+  ONGOING,
+  RECOVERED,
+}
+
+const getPersistentAlertsByTimeRange = async <TSearchRequest extends ESSearchRequest>({
+  start,
+  end,
+  ruleId,
+  ruleDataClientReader,
+}: GetAlertsByTimeRangeHelperOpts) => {
+  // persistent alerts only create new alerts so query for all alerts within the time
+  // range and treat them as NEW
+  const request = getQueryByTimeRange(start, end, ruleId);
+  const response = (await ruleDataClientReader.search(request)) as ESSearchResponse<
+    AlertDocument,
+    TSearchRequest
+  >;
+
+  return {
+    new: getHitsWithCount(response),
+    ongoing: {
+      count: 0,
+      alerts: [],
+    },
+    recovered: {
+      count: 0,
+      alerts: [],
+    },
+  };
+};
+
+const getLifecycleAlertsByTimeRange = async ({
+  start,
+  end,
+  ruleId,
+  ruleDataClientReader,
+}: GetAlertsByTimeRangeHelperOpts) => {
+  const requests = [
+    getQueryByTimeRange(start, end, ruleId, AlertTypes.NEW),
+    getQueryByTimeRange(start, end, ruleId, AlertTypes.ONGOING),
+    getQueryByTimeRange(start, end, ruleId, AlertTypes.RECOVERED),
+  ];
+
+  const responses = await Promise.all(
+    requests.map((request) => ruleDataClientReader.search(request))
+  );
+
+  return {
+    new: getHitsWithCount(responses[0]),
+    ongoing: getHitsWithCount(responses[1]),
+    recovered: getHitsWithCount(responses[2]),
+  };
+};
+
+const getQueryByTimeRange = (start: Date, end: Date, ruleId: string, type?: AlertTypes) => {
+  // base query filters the alert documents for a rule by the given time range
+  let filter: QueryDslQueryContainer[] = [
+    {
+      range: {
+        [TIMESTAMP]: {
+          gte: start.toISOString(),
```

Comment:
should one of these be exclusive (`lt` instead of `lte`)?  Let's say someone's doing bucketing of these ranges, and one of the alerts is right at one of the bucket ranges.  It would be counted in both buckets.  Probably exceeding rare, but seems like something we should do.  Not sure which one would be exclusive though.  Probably the end.

Reply:
I think this should be rare but I can update! The `@timestamp`, `kibana.alert.start` and `kibana.alert.end` for lifecycle alerts when they are set are all set to the `task.startedAt` value for that execution.